core/vm/consortium_precompiled_contracts.go

```diff
@@ -245,42 +245,41 @@ func (c *consortiumValidatorSorting) Run(input []byte) ([]byte, error) {
 	if len(validators) != len(weights) {
 		return nil, errors.New("balances and validators length mismatched")
 	}
-	sortValidators(validators, weights)
+	validators, weights = sortValidators(validators, weights)
 
 	log.Debug("Precompiled sorting", "validators", validators, "weights", weights)
 
 	return method.Outputs.Pack(validators)
 }
 
-func sortValidators(validators []common.Address, weights []*big.Int) {
-	if len(validators) < 2 {
-		return
-	}
+type SortableValidators struct {
+	validators []common.Address
+	weights    []*big.Int
+}
 
-	left, right := 0, len(validators)-1
+func (s *SortableValidators) Len() int {
+	return len(s.validators)
+}
 
-	pivot := rand.Int() % len(validators)
+func (s *SortableValidators) Less(i, j int) bool {
+	cmp := s.weights[i].Cmp(s.weights[j])
+	addrsCmp := big.NewInt(0).SetBytes(s.validators[i].Bytes()).Cmp(big.NewInt(0).SetBytes(s.validators[j].Bytes())) > 0
+	return cmp > 0 || (cmp == 0 && addrsCmp)
+}
 
-	validators[pivot], validators[right] = validators[right], validators[pivot]
-	weights[pivot], weights[right] = weights[right], weights[pivot]
+func (s *SortableValidators) Swap(i, j int) {
+	s.validators[i], s.validators[j] = s.validators[j], s.validators[i]
+	s.weights[i], s.weights[j] = s.weights[j], s.weights[i]
+}
 
-	for i := range validators {
-		cmp := weights[i].Cmp(weights[right])
-		addrsCmp := big.NewInt(0).SetBytes(validators[i].Bytes()).Cmp(big.NewInt(0).SetBytes(validators[right].Bytes())) > 0
-		if cmp > 0 || (cmp == 0 && addrsCmp) {
-			validators[left], validators[i] = validators[i], validators[left]
-			weights[left], weights[i] = weights[i], weights[left]
-			left++
-		}
+func sortValidators(validators []common.Address, weights []*big.Int) ([]common.Address, []*big.Int) {
```

Comment:
I think we don't need to return addresses and balances here, we only change the order of elements in the slice and does not append the slice, so any changes we make to slice inside this function result in the change in caller provided's slice.